src/provider/states/pod/starting.rs

```diff
@@ -72,32 +64,19 @@ async fn start_service_units(
 
     for (container_key, container_handle) in pod_handle.unwrap_or_default() {
         let systemd_service = &container_handle.systemd_service;
+        let service_unit = &container_handle.service_unit;
 
-        if systemd_service.is_running().await? {
-            debug!(
-                "Unit [{}] for service [{}] is already running. Skip startup.",
-                systemd_service.file(),
-                &pod_state.service_name
-            );
-        } else {
-            let service_unit = &container_handle.service_unit;
-
+        if systemd_service.service_state().await? == ServiceState::Created {
```

Comment:
This condition is not met in a restart loop.

`ServiceState::Created` means that the service was created but never started. When the service is started then `systemd_service.service_state()` returns `ServiceState::Started`. It will not leave this state until it is terminated (forever) and it will never return to `ServiceState::Created`. So whatever happens while restarting, `systemd_service.service_state()` always returns `ServiceState::Started`.